CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+
+| Setting Type | CLI Flag | Env Var | Config File | Recommended Use |
+| ----- | ----- | ----- | ----- | ----- |
+| Frequently changed | Yes | Optional | Optional | Expect different values for different executions. |
+| Secret or credential | No | Yes | Optional | Use Env for credential by default. Optionally use the config file. |
+| Static or persistent | No | Optional | Yes | Use config files |
+| Tweakable default | Yes | Optional | Yes | Use CLI for override, config for base; optionally env |
+
+### Command Input
+
+#### Interactive & non-interactive
+
+Keep in mind that users may run the backplane commands in a script. If you are asking for interactive input, always provide an option to accept the input non-interactively, eg, by flags or environment variables.
+
+Example:  
+The elevate command by default accept the reason non-interactively:
+
+```shell
+ocm backplane elevate <reason> -- get po -A
+```
+
+It only prompts interactive input when specified with \`-n\`.
+
+```shell
+ocm backplane elevate -n -- get po -A
+```
+
+### Command Output
+
+#### Format
+
+If the command is to retrieve an info for the user, print the output to stdout. Be friendly to command line tools like grep and awk.  
+Optionally, provide an option to output json format for programmatically processing.
+
+Example:  
+`managedjob` provides an option to output raw response from backplane-api, which is a json format.
+
+```shell
+ocm-backplane managedjob create <script name> --raw
+```
+
+Command `cloud credentials` provide options to output in different formats.
+
+```go
+ocm-backplane cloud credentials
+-o, --output string   Format the output of the credentials response. One of text|json|yaml|env (default "text")
+```
+
+#### Verbose
+
+Provide verbose output.  The backplane users are mostly with technical backgrounds.
+Share more information for users to debug.
+
+**Debug**
+Detailed progress of internal processes
+
+**Info**
+Major steps in a multi-stage process
+Configuration details being used
+
+**Warn**
+Deprecated features being used
+Non-critical configuration issues
+
+**Error**
+Invalid input or arguments
+Fatal application errors
+
+Example:
+```
+DEBU  Running Login Command
+DEBU  Checking Backplane Version
+WARN  Your Backplane CLI is not up to date. Please run the command 'ocm backplane upgrade' to upgrade to the latest version  Current version=0.1.44 Latest version=0.1.47
+DEBU  Extracting Backplane configuration
+ERRO  failed to create OCM connection: please ensure you are logged into OCM by using the command "ocm login \--url $ENV"
+```
+
+### Error Handing
+
+Throw errors, don't hide errors.
+
+Poor: This one hides the error, the user may be frustrated on what went wrong.
+
+```go
+if err != nil {
+  return fmt.Errorf("can't find shard url")
+}
+```
+
+Good: This one returns the underlying error, so the user can troubleshoot easier.
+
+```go
+if err != nil {
+  return fmt.Errorf("can't find shard url: %v", err)
+}
+```
+
```

Comment:
Should we also include testing guidelines in this doc for the contributor?  for example:
- unit test coverage
- end-to-end test when applicable